### DISCUSSION TITLE
how to remove ssl on vercel account?

### DISCUSSION THREAD

**Discussion by m0x61h0x64i** [vercel/community#1005] [Help] [answered]
hello, i am using cloudflare and i have to use flexible mode instead of full mode because its the only way that i can run my other service on the domain (i dont have command line to install ssl on that service and they dont offer ssl), so when im using full mode vercel works but when i switch to flexible mode it dosent work because i have to remove the ssl and let the cloudflare to use his own ssl between browser and vercel server, i already tried removing ssl on the vercel panel on domains tab but that didnt work!
![image](https://user-images.githubusercontent.com/109550653/197727771-39bbbf40-b513-4a73-b09d-1df6aa8e1276.png)

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out @m0x31h0x64i! When using Cloudflare Proxy with a Vercel site, you need to use full encryption mode as described in the [domain proxy guide](https://vercel.com/guides/using-cloudflare-with-vercel#with-proxy). My understanding is that flexible mode results in a redirect loop because of cert checking processes (but that isn't my area of expertise, so my understanding may be faulty). There is no way to use an unsecured domain with Vercel.

If you share more details about what you need to accomplish, maybe we can come up with another way to make it work. [Middleware](https://vercel.com/docs/concepts/functions/edge-middleware) and [rewrites](https://vercel.com/docs/project-configuration#project-configuration/rewrites) are common solutions to similar problems. But there are likely other options depending on what you want to build.
